### ISSUE TITLE
SQL Translation Error - Transact-SQL outer joins

### ISSUE THREAD

**Issue by andyedison** [jOOQ/jOOQ#7300]
### SQL translator incorrectly translates this query:

```sql
select p.* 
from people p, people_info pi
where p.id *= pi.id
```

### Parser output:

```sql
Identifier expected: [3:13] ...ct p.* 
from people p, people_info pi
where p.id *[*]= pi.id
```

SAP (Sybase) ASE has and still supports transact-sql outer joins (star joins, *=). It's a tough problem, but a way to convert these to say SQL Server would be very valuable. 

[ASE Documentation on tsql outer joins](https://help.sap.com/viewer/b65d6a040c4a4709afd93068071b2a76/16.0.2.6/en-US/aa2dcdc9bc2b1014b423b992927db174.html?q=Transact-SQL%20outer%20joins)

**Comment by lukaseder** [comment]
Thank you very much for your report. It definitely makes sense to parse that syntax. Older SQL Server versions also still support it. Oracle (and curiously: the latest DB2 versions) support a `(+)` operator, which does the same thing.

It's probably not too hard to transform between

- `a *= b` and `a = b(+)`
- `a =* b` and `a(+) = b`

Would be really interesting to transform these to ANSI joins, too, although the transformation is much more complicated. Will think about it.

**Comment by lukaseder** [comment]
The parser has been enhanced to recognise these syntaxes: https://github.com/jOOQ/jOOQ/commit/c5f1c85f27232dea2a80a6913f54c5b9086d27c1

Right now, the internal representation of `*=` and `=*` is using the existing support for the Oracle syntax.

Emulation of the syntax / translation to ANSI join will be implemented separately: #7312

Thanks again for reporting!
